homeassistant/components/image_processing/facebox.py

```diff
@@ -75,19 +79,41 @@ def parse_faces(api_faces):
     return known_faces
 
 
-def post_image(url, image):
+def post_image(url, username, password, image):
     """Post an image to the classifier."""
     try:
         response = requests.post(
             url,
+            auth=requests.auth.HTTPBasicAuth(username, password),
             json={"base64": encode_image(image)},
-            timeout=TIMEOUT
             )
+        if response.status_code == HTTP_UNAUTHORIZED:
+            _LOGGER.error("AuthenticationError on %s", CLASSIFIER)
+            return None
         return response
     except requests.exceptions.ConnectionError:
         _LOGGER.error("ConnectionError: Is %s running?", CLASSIFIER)
 
 
+def teach_file(url, username, password, name, file_path):
+    """Teach the classifier a name associated with a file."""
+    try:
+        with open(file_path, 'rb') as open_file:
+            response = requests.post(
+                url,
+                auth=requests.auth.HTTPBasicAuth(username, password),
```

Comment:
What if username or password is None?

Reply:
They are by default and its OK

Reply:
Don't send in auth if username and password are `None`, that triggers a deprecation warning.

```python
kwargs = {}

if username:
    kwargs['auth'] = requests.auth.HTTPBasicAuth(username, password)

response = request.post(url, **kwargs)
```